src/plugins/unified_search/public/plugin.ts

```diff
@@ -73,16 +73,27 @@ export class UnifiedSearchPublicPlugin
     setIndexPatterns(dataViews);
     const autocompleteStart = this.autocomplete.start();
 
-    const SearchBar = createSearchBar({
-      core,
-      data,
-      storage: this.storage,
-      usageCollection: this.usageCollection,
-      isScreenshotMode: Boolean(screenshotMode?.isScreenshotMode()),
-      unifiedSearch: {
-        autocomplete: autocompleteStart,
-      },
-    });
+    /*
+     *
+     *  unifiedsearch uses global data service to create stateful search bar.
+     *  This function helps in creating a search bar with different instances of services
+     *  so that it can be easy to use multiple stateful searchbars in the single applications
+     *
+     * */
+    const getCustomSearchBar = (deps?: Partial<StatefulSearchBarDeps>) =>
+      createSearchBar({
+        core,
+        data,
+        storage: this.storage,
+        usageCollection: this.usageCollection,
+        isScreenshotMode: Boolean(screenshotMode?.isScreenshotMode()),
+        unifiedSearch: {
+          autocomplete: autocompleteStart,
+        },
+        ...(deps ?? {}),
+      });
```

Comment:
Can you explain a bit why we are doing this change? Which is the use case?

Reply:
If this is only for ovewriting the data service wouldn't it better to just pass the service here? Because now the consumer can ovewrite everything and I don't think is something we want tbh

Reply:
Thank @stratoula  for taking a look.

Yes, your observation is correct that, we need to pass a custom instance of data services, instead of global singleton instance. This will help us get a stateful `SearchBar` via navigation plugin which does not affect the `GlobalSearchBar` as used by all apps.


>Because now the consumer can ovewrite everything and I don't think is something we want tbh
You are correct and I added to be as generic as well possible but if this is something you think can create issues, we can certainly discuss the changes that can be done to mitigate those issues.

>If this is only for ovewriting the data service wouldn't it better to just pass the service here?
What do you mean by this? you means that instead of all the `deps`, I just pass the data service and that is it?

Reply:
@stratoula , PR is ready to be reviewed again based on the inputs provided by you.